### PR TITLE
feat(992/us8): T092b FR-027 display-format column resolution in DetailList

### PR DESCRIPTION
**Pull request by natechadwick-intsof**
## Summary

Closes the **complete US8 / Phase 10 follow-on queue** — five tasks that close feature gaps surfaced by `/speckit.analyze`:

- **T092b** — FR-027 display-format column resolution in `DetailList`
- **T092c** — Edge Cases #3 concurrent rename/move 409 surfacing
- **T092d** — Edge Cases #7 cross-frame session + CSRF cert
- **T092e** — Edge Cases #11 network failure mid-action cert
- **T029b** — FR-019a CI-gate artifact-grep for `finder.jsp` navigation entries

**Spec 992 is feature-complete on this branch**: 32/32 capability matrix rows Done, US8 sub-PRs merged, Edge Cases #3 / #7 / #11 closed, FR-019a measurable gate in place.

## T092b — FR-027 display-format column resolution

Adds an optional `displayFormat` prop to the modern Content Explorer's `DetailList` so the list honours host-supplied column ordering + selection. Default columns (Name + Type + Path) are preserved when `displayFormat` is absent or empty.

**Implementation:** `DetailColumnId` union (7 ids), `DetailDisplayFormat` interface, 3 pure helpers (`resolveDisplayFormatColumns`, `renderDisplayFormatCell`, `columnHeaderLabel`), column-aware `<thead>` / `<tbody>` using `data-testid="detail-col-header-{c}"` / `"detail-cell-{c}-{id}"` selectors, 4 new i18n keys.

**Tests:** 8 new Vitest tests in `DetailList.test.tsx`. 6 pre-existing mock responses corrected to wire shape `{ PagedItemList: { childrenInPage, childrenCount, startIndex } }`.

## T092c — Edge Cases #3 concurrent rename/move 409 surfacing

Closes the client half of Edge Cases #3: surface HTTP status from the moveItem failure through the paste summary so the second of two concurrent rename/move clients sees a 409 + clear error message (no silent overwrite, no data corruption).

**Implementation:** `ClipboardPasteResultItem.status?: number`, status-extraction branch in `pasteClipboardItems`, `data-conflict="true"` on failure rows in `ClipboardPanel`.

**Tests:** 4 new Vitest tests (3 in `clipboardApi.test.ts` + 1 in `ClipboardPanel.test.tsx`). Playwright spec `us8-edge-cases-concurrent-move.spec.js`.

## T092d — Edge Cases #7 cross-frame session + CSRF cert

Certifies the existing CSRF contract handles cross-frame session rotation. **No client code change required** — `getCsrfToken` already reads the global fresh per call and `client.ts#buildHeaders` already calls `getCsrfToken` per request.

**Tests:** 4 new Vitest tests in `WebUI/src/test/ts/api/csrf.test.ts` (NEW). Playwright spec `us8-edge-cases-cross-frame.spec.js`.

## T092e — Edge Cases #11 network failure mid-action cert

Certifies the paste-flow handles network failure mid-action with no data corruption, recoverable error surfacing, and re-auth + retry without a hard refresh. **No client code change required** — `pasteClipboardItems`' `Promise.allSettled` + per-item message contract already satisfies the requirement.

**Tests:** 4 new Vitest tests in `clipboardApi.test.ts` (T092e describe block). Playwright spec `us8-edge-cases-network-failure.spec.js`.

## T029b — FR-019a CI-gate artifact-grep

Closes the final open task in the spec 992 follow-on queue. Adds a CI-gate artifact-grep that asserts the production-built WebUI WAR contains zero `finder.jsp` navigation-entry references in `cm/app/webmgt.jsp` (the modern Track B shell hard-cut in US6 PR #1390), proving no production classic fallback ships.

**Implementation:**
- `scripts/verify-no-finder-jsp-references.sh` (POSIX) + `.bat` (Windows): regex matched against the navigation-entry forms (`<jsp:include page="includes/finder.jsp">`, `<%@include file="includes/finder.jsp">`), explicitly carving out (a) the `finder_js.jsp` shared-library include (required for non-Finder functionality: `PercComponentWrapper`, `PercViewReadyManager`, `PercPathService`, etc.) and (b) `cm/pages/app/webmgt.jsp` Track A path (deferred to the Track A migration workstream).
- Multi-line JSP comment stripping via `perl -0777 -pe 's/<%--.*?--%>//gs'` so the US6 cutover comment doesn't false-positive.
- `scripts/test-verify-no-finder-jsp-references.sh`: paired shell self-test (4 cases: PASS on current tree, FAIL on re-introduced `<jsp:include>`, FAIL on `<%@include>`, PASS on `finder_js.jsp` shared-lib include carve-out).
- `WebUI/src/test/ts/scripts/verify-no-finder-jsp-references.test.ts`: Vitest spec — load-bearing CI assertion (2 tests: script file present, PASS on current tree).

## Docs

- `specs/992-react-content-explorer/checklists/edge-cases-coverage.md` (NEW) — Edge Cases ↔ test anchor map; Edge Cases #3 / #7 / #11 flipped from `gap` / `partial` to `covered` per T092c / T092d / T092e.
- `specs/992-react-content-explorer/tasks.md` — T029b + T092b + T092c + T092d + T092e entries marked done with evidence.
- `scripts/README.md` — new entry for the gate + Windows counterpart + Spec ref T029b.
- `docs/ai-generated/code-reviews/992-react-content-explorer-us8-t029{b,c,d,e,b}-erlang.md` — pre-push Erlang reviews (all APPROVE).

## Constitutional compliance

| Constraint | T029b | T092b | T092c | T092d | T092e |
|------------|-------|-------|-------|-------|-------|
| I (no invariants violated) | ✅ | ✅ | ✅ | ✅ | ✅ |
| II (no invented APIs) | ✅ N/A | ✅ | ✅ | ✅ N/A | ✅ N/A |
| III (behavioral tests) | ✅ 2 + 4 self | ✅ 8 | ✅ 4 | ✅ 4 | ✅ 4 |
| V (Plan / Complexity) | ✅ focused | ✅ focused | ✅ focused | ✅ focused | ✅ focused |
| VII (format checks) | ✅ tsc + 2/2 vitest | ✅ tsc + 13/14 | ✅ tsc + 8/8 + 11/11 | ✅ tsc + 4/4 | ✅ tsc + 12/12 |
| E (no residuals) | ✅ FR-019a gate | ✅ FR-027 | ✅ #3 | ✅ #7 | ✅ #11 |

## Pre-push Erlang reviews

`docs/ai-generated/code-reviews/992-react-content-explorer-us8-t092b-erlang.md`
`docs/ai-generated/code-reviews/992-react-content-explorer-us8-t092c-erlang.md`
`docs/ai-generated/code-reviews/992-react-content-explorer-us8-t092d-erlang.md`
`docs/ai-generated/code-reviews/992-react-content-explorer-us8-t092e-erlang.md`
`docs/ai-generated/code-reviews/992-react-content-explorer-us8-t029b-erlang.md`

```
T029b: RECOMMENDATION approve / 0 blocking / 4 informational
T092b: RECOMMENDATION approve / 0 blocking / 3 informational
T092c: RECOMMENDATION approve / 0 blocking / 3 informational
T092d: RECOMMENDATION approve / 0 blocking / 3 informational
T092e: RECOMMENDATION approve / 0 blocking / 3 informational
PORTABILITY CHECK:  0 unix-only / 0 windows-only paths (all)
FAILS (any):        no (all)
```

### CODE REVIEW

File: scripts/verify-no-finder-jsp-references.bat [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+REM rationale and carve-outs). findstr does NOT have a portable way
+REM to strip JSP comments inline, so we use a small PowerShell one-liner
+REM to strip <%-- ... --%> blocks before the findstr invocation.
+for %%F in ("%TARGET_JSP%") do (
+    if not exist "%%~fF" (
+        echo   FAIL: target JSP does not exist: %%~fF 1>&2
+        set "FAIL=1"
+    ) else (
+        for /f "usebackq tokens=*" %%M in (
+            `powershell -NoProfile -Command "Get-Content -LiteralPath '%%~fF' | ForEach-Object { $_ -replace '<%--.*?--%>','' } | Select-String -Pattern '<jsp:include[ \t]+page=\"includes/finder\.jsp|<%%@include[ \t]+file=\"includes/finder\.jsp'"`
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply scripts/verify-no-finder-jsp-references.bat]: **Mitigation (commit `a5b66674ed`):** Replaced the per-line PowerShell pipeline with `[System.IO.File]::ReadAllText` (slurps whole file, equivalent to POSIX `perl -0777`) + `[regex]::Replace(content, '(?s)<%--.*?--%>', '')` with the `(?s)` inline flag (makes `.` match newlines, equivalent to the `/s` flag in the .sh perl regex). Verified the .sh regex semantics via the shell self-test (4/4 cases passing) — the PowerShell regex is the equivalent inline-flag form. The misleading "Use findstr" comment was also updated to reflect PowerShell as the actual parsing engine (see also the companion Sp6JR comment in this thread).

File: scripts/verify-no-finder-jsp-references.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+target_jsp="WebUI/src/main/webapp/cm/app/webmgt.jsp"
+
+echo "==> checking ${target_jsp} for finder.jsp navigation entries"
+for path in $target_jsp; do
+    if [ ! -f "$path" ]; then
+        echo "  FAIL: target JSP does not exist: $path" >&2
+        fail=1
+        continue
+    fi
+    matches=$(perl -0777 -pe 's/<%--.*?--%>//gs' "$path" | \
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply scripts/verify-no-finder-jsp-references.sh:84]: **Mitigation (commit `a5b66674ed`):** Added an early `command -v perl >/dev/null 2>&1` check at the top of the gate (right after the repo-root `cd`). If perl is not on PATH, the gate now fails loudly with `verify-no-finder-jsp-references: FAIL (perl is required but not on PATH)` and exit 1, instead of silently passing due to the `|| true` at the end of the pipeline. perl is standard on Linux + macOS dev hosts and on the GitHub Actions runners (ubuntu-latest / windows-latest / macos-latest all ship perl by default), so this should not trigger in normal CI; the check exists for environments where perl has been deliberately stripped or unavailable.

File: scripts/test-verify-no-finder-jsp-references.sh [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+# (cm/app/webmgt.jsp is hard-cut per US6). The FAIL cases assert the
+# script returns 1 when (a) a navigation entry is re-introduced and
+# (b) the regex doesn't accidentally match the shared-library
+# finder_js.jsp include (which is intentionally allowed as a
+# carve-out, see the .sh header comment).
+#
+# Any scratch files/directories this test creates are removed on exit
+# (via trap, so cleanup still runs on failure/interrupt) and are never
+# committed.
+set -u
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply scripts/test-verify-no-finder-jsp-references.sh]: **Mitigation (commit `a5b66674ed`):** Replaced `set -u` with `set -eu`. POSIX-portable: dropped the `set -o pipefail` (bash extension not supported by dash). The test's failure-mode pipelines (cp backup, heredoc append) are now caught by `set -e`; the intentional control-flow pipelines (`if sh "$script" ...; then ... else ... fi`) are unaffected because `set -e` only triggers on unhandled non-zero returns. Re-ran the self-test: all 4 cases still pass (PASS on current tree; FAIL on re-introduced <jsp:include>; FAIL on <%@include>; PASS on finder_js.jsp shared-lib include carve-out).

File: scripts/verify-no-finder-jsp-references.bat [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+set "REPO_ROOT=%~dp0.."
+pushd "%REPO_ROOT%" >nul
+
+set "FAIL=0"
+set "TARGET_JSP=WebUI\src\main\webapp\cm\app\webmgt.jsp"
+
+echo ==^> checking %TARGET_JSP% for finder.jsp navigation entries
+
+REM Use findstr (Windows port of grep) with regex via /R. The pattern
```
[the remark this answers is not in the document]
Comment (natechadwick-intsof) [reply scripts/verify-no-finder-jsp-references.bat]: **Mitigation (commit `a5b66674ed`):** Removed the misleading "Use findstr (Windows port of grep)" comment and replaced it with an accurate implementation note that names PowerShell as the actual parsing engine (with the [regex]::Replace + (?s) flag equivalent to the POSIX `perl -0777 -pe /s` step in the .sh). The same commit also fixes the underlying PowerShell pipeline to use `[System.IO.File]::ReadAllText` + `(?s)<%--.*?--%>` so the multi-line JSP comment at cm/app/webmgt.jsp:330-340 is stripped before the navigation-entry regex runs (see companion Sp6JA thread in this PR for the regex fix details).

**Review by vijaya-boddipudi — APPROVED** [review]
